go/plugins/compat_oai/openai/openai_live_test.go

```diff
@@ -0,0 +1,221 @@
+			Value float64
+			Over  float64
+		},
+		) (float64, error) {
+			return math.Pow(input.Value, input.Over), nil
+		},
+	)
+
+	t.Log("openai plugin initialized")
+
+	t.Run("embedder", func(t *testing.T) {
+		embedder := oai.Embedder(g, "text-embedding-3-small")
+		res, err := ai.Embed(ctx, embedder, ai.WithEmbedText("yellow banana"))
+		if err != nil {
+			t.Fatal(err)
+		}
+		out := res.Embeddings[0].Embedding
+		// There's not a whole lot we can test about the result.
+		// Just do a few sanity checks.
+		if len(out) < 100 {
+			t.Errorf("embedding vector looks too short: len(out)=%d", len(out))
+		}
+		var normSquared float32
+		for _, x := range out {
+			normSquared += x * x
+		}
+		if normSquared < 0.9 || normSquared > 1.1 {
+			t.Errorf("embedding vector not unit length: %f", normSquared)
+		}
+	})
+
+	t.Run("basic completion", func(t *testing.T) {
+		t.Log("generating basic completion response")
+		resp, err := genkit.Generate(ctx, g,
+			ai.WithPromptText("What is the capital of France?"),
+		)
+		if err != nil {
+			t.Fatal("error generating basic completion response: ", err)
+		}
+		t.Logf("basic completion response: %+v", resp)
+
+		out := resp.Message.Content[0].Text
+		if !strings.Contains(strings.ToLower(out), "paris") {
+			t.Errorf("got %q, expecting it to contain 'Paris'", out)
+		}
+
+		// Verify usage statistics are present
+		if resp.Usage == nil || resp.Usage.TotalTokens == 0 {
+			t.Error("Expected non-zero usage statistics")
+		}
+	})
+
+	t.Run("streaming", func(t *testing.T) {
+		var streamedOutput string
+		chunks := 0
+
+		final, err := genkit.Generate(ctx, g,
+			ai.WithPromptText("Write a short paragraph about artificial intelligence."),
+			ai.WithStreaming(func(ctx context.Context, chunk *ai.ModelResponseChunk) error {
+				chunks++
+				for _, content := range chunk.Content {
+					streamedOutput += content.Text
+				}
+				return nil
+			}))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Verify streaming worked
+		if chunks <= 1 {
+			t.Error("Expected multiple chunks for streaming")
+		}
+
+		// Verify final output matches streamed content
+		finalOutput := ""
+		for _, content := range final.Message.Content {
+			finalOutput += content.Text
+		}
+		if streamedOutput != finalOutput {
+			t.Errorf("Streaming output doesn't match final output\nStreamed: %s\nFinal: %s",
+				streamedOutput, finalOutput)
+		}
+
+		t.Logf("streaming response: %+v", finalOutput)
+	})
+
+	t.Run("tool usage with basic completion", func(t *testing.T) {
+		resp, err := genkit.Generate(ctx, g,
+			ai.WithPromptText("what is a gablorken of 2 over 3.5?"),
+			ai.WithTools(gablorkenTool))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		out := resp.Message.Content[0].Text
+		const want = "12.25"
+		if !strings.Contains(out, want) {
+			t.Errorf("got %q, expecting it to contain %q", out, want)
+		}
+
+		t.Logf("tool usage with basic completion response: %+v", out)
+	})
+
+	t.Run("tool usage with streaming", func(t *testing.T) {
```

Comment:
It's flaky, just like the function calling test with basic completion.

Reply:
Same comment: https://github.com/firebase/genkit/pull/2440#discussion_r2040361890